diff --git a/pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh b/pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh
index 2d13a5e8137..12a72b9053b 100755
--- a/pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh
+++ b/pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh
@@ -329,8 +329,10 @@ if [[ -n $flake ]]; then
         if [[ -z $hostname ]]; then
             hostname=default
         fi
+        configName="\"$hostname\""
         flakeAttr="nixosConfigurations.\"$hostname\""
     else
+        configName="\"$flakeAttr\""
         flakeAttr="nixosConfigurations.\"$flakeAttr\""
     fi
 fi
@@ -478,6 +480,9 @@ if [ -z "$rollback" ]; then
         fi
         copyToTarget "$pathToConfig"
         targetHostCmd nix-env -p "$profile" --set "$pathToConfig"
+        if [[ -n $flake ]]; then
+            targetHostCmd @storeNewGeneration@ "$profile" "$pathToConfig" "$configName" "${lockFlags[@]}" >&2
+        fi
     elif [[ "$action" = test || "$action" = build || "$action" = dry-build || "$action" = dry-activate ]]; then
         if [[ -z $flake ]]; then
             pathToConfig="$(nixBuild '<nixpkgs/nixos>' -A system -k "${extraBuildFlags[@]}")"
