diff --git a/hosts/hermes/hardware-configuration.nix b/hosts/hermes/hardware-configuration.nix
index 5e5fdf9..738de1b 100644
--- a/hosts/hermes/hardware-configuration.nix
+++ b/hosts/hermes/hardware-configuration.nix
@@ -26,7 +26,7 @@
   swapDevices = [{ device = "/dev/pool/swap"; }];
 
   hardware.asahi.useExperimentalGPUDriver = true;
-  hardware.asahi.experimentalGPUInstallMode = "replace";
+  hardware.asahi.experimentalGPUInstallMode = "overlay";
 
   powerManagement.cpuFreqGovernor = lib.mkDefault "ondemand";
 }
