diff --git a/flake.nix b/flake.nix
index cf8fbb7..cad4e0a 100644
--- a/flake.nix
+++ b/flake.nix
@@ -286,14 +286,6 @@
           inherit (modules) laptop linux-builder personal;
         };
       }
-      {
-        host = "hermes";
-        hostSuffix = "-nixos";
-        user = "enzime";
-        system = "aarch64-linux";
-        modules =
-          builtins.attrNames { inherit (modules) laptop personal sway; };
-      }
       {
         host = "phi";
         hostSuffix = "-nixos";
